include/ipmitool/cssipmi.h

```diff
@@ -0,0 +1,263 @@
+    IPMI Strcutures
+**************************************************************************/
+
+typedef struct _IPMISDRHeader 
+{
+  /* record ID of the SDR */
+  unsigned short recordID;
+  /* SDR version info */
+  unsigned char sdrVer;
+  /* type of SDR record  */
+  unsigned char recordType;
+  /* SDR record lenght */
+  unsigned char recordLength;
+}CSSD_PACKED IPMISDRHeader;
+
+typedef struct _IPMISDRType1 
+{
+  unsigned char ownerID;
+  unsigned char ownerLUN;
+  unsigned char sensorNum;
+  unsigned char entityID;
+  unsigned char entityInstance;
+  unsigned char sensorInit;
+  unsigned char sensorCaps;
+  unsigned char sensorType;
+  unsigned char readingType;
+  unsigned short triggerLTRMask; 
+  unsigned short triggerUTRMask;
+  unsigned short readingMask;     
+  unsigned char units1;
+  unsigned char units2;
+  unsigned char units3;
+  unsigned char linearization;
+  unsigned char m;
+  unsigned char tolerance;
+  unsigned char b;
+  unsigned char accuracy;
+  unsigned char accuracyExp;
+  unsigned char rbExp;
+  unsigned char analogChars;
+  unsigned char nominalReading;
+  unsigned char normalMax;
+  unsigned char normalMin;
+  unsigned char sensorMax;
+  unsigned char sensorMin;
+  unsigned char upperNR;
+  unsigned char upperC;
+  unsigned char upperNC;
+  unsigned char lowerNR;
+  unsigned char lowerC;
+  unsigned char lowerNC;
+  unsigned char positiveHystersis;
+  unsigned char negativeHysterisis;
+  unsigned char reserved1;
+  unsigned char reserved2;
+  unsigned char OEM;
+  unsigned char typeLengthCode;
+  unsigned char sensorName[16]; 
+}CSSD_PACKED IPMISDRType1; 
+
+typedef struct _IPMISDRType2 
+{
+  unsigned char ownerID;
+  unsigned char ownerLUN;
+  unsigned char sensorNum;
+  unsigned char entityID;
+  unsigned char entityInstance;
+  unsigned char sensorInit;
+  unsigned char sensorCaps;
+  unsigned char sensorType;
+  unsigned char readingType;
+  unsigned short triggerLTRMask;  
+  unsigned short triggerUTRMask;
+  unsigned short readingMask;     
+  unsigned char units1;
+  unsigned char units2;
+  unsigned char units3;
+  unsigned char recordSharing1;
+  unsigned char recordSharing2;
+  unsigned char positiveHystersis;
+  unsigned char negativeHysterisis;
+  unsigned char reserved1;
+  unsigned char reserved2;
+  unsigned char reserved3;
+  unsigned char OEM;
+  unsigned char typeLengthCode;
+  unsigned char sensorName[16];   
+}CSSD_PACKED IPMISDRType2; 
+
+typedef struct _IPMISDRType8 
+{
+  unsigned char containerEntityID;
+  unsigned char containerEntityInstance;
+  unsigned char flags;
+  unsigned char containedEntityID1R1;     
+  unsigned char containedEntityInstance1R1;
+  unsigned char containedEntityID2R1;     
+  unsigned char containedEntityInstance2R1; 
+  unsigned char containedEntityID3R2;     
+  unsigned char containedEntityInstance3R2; 
+  unsigned char containedEntityID4R2;     
+  unsigned char containedEntityInstance4R2; 
+}CSSD_PACKED IPMISDRType8; 
```

Comment:
Please DON'T repeat standard IPMI definitions. All this stuff is already defined elsewhere in ipmitool. DELL isn't the only company using standard SDRs and other IPMI entities. You can't just take a chunk of code from a totally strange project and just apply it on top of ipmitool. That results, putting it politely, in a very unmaintainable and poor quality project.

I think I'll just stop here and wait until you re-submit this PR as a series of concise, clean, and up-to-standards patches that only add what is actually needed to support your hardware within the existing ipmitool framework.